### PR TITLE
[BEAM-3342] CBT `Read` IO Connector Wrapper

### PR DESCRIPTION
**Pull request by mf2199**
[CBT connector: Expanding `bigtableio` with `read` functionality.]

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by chamikaramj** [comment]
Retest this please

**Comment by chamikaramj** [comment]
Run Python PostCommit

**Comment by chamikaramj** [comment]
Run Python PreCommit

**Comment by chamikaramj** [comment]
Retest this please

**Comment by chamikaramj** [comment]
Run Python PostCommit

**Comment by chamikaramj** [comment]
Please ping here if any tests have to be re-triggered.

**Comment by chamikaramj** [comment]
Can you please move these modules (including the BigTable sink) to io/gcp/experimental.

We should leave this there till we have have proper integration and performance tests in place to be confident about the production readiness of this the BigTable source and sink.

**Comment by chamikaramj** [comment]
Pre-commit failure seems to be related.

**Comment by mf2199** [comment]
Run Python PreCommit

**Comment by aaltay** [comment]
retest this please

**Comment by mf2199** [comment]
@chamikaramj @aaltay At this point, the errors point to missing arguments that were made so by design, e.g.:
```
15:41:25 E       ValueError: Pipeline has validations errors: 
15:41:25 E       Missing required option: project.
15:41:25 E       Missing required option: region.
15:41:25 E       Missing GCS path option: temp_location.
15:41:25 E       Missing GCS path option: staging_location.
```
[from "Run Python PreCommit" log]

These were left out purposely, as they may contain non-public information and subject to change. 

Any ideas?

**Comment by chamikaramj** [comment]
Retest this please

**Comment by chamikaramj** [comment]
Lots were not available anymore. Re-triggerred.

File: sdks/python/apache_beam/io/gcp/experimental/bigtableio_read_it_test.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+          '%d out of %d rows were read successfully.',
+          final_count,
+          self.options['row_count'])
+
+    logging.info('DONE!')
+
+
+def parse_commane_line_arguments():
+  parser = argparse.ArgumentParser()
+
```
Comment (chamikaramj) [inline sdks/python/apache_beam/io/gcp/experimental/bigtableio_read_it_test.py:123]: You do not need to set these up manually. Please see the way our other integration tests are setup. You can either configure a new Jenkins test suite or add integration tests to existing Python post-commit tests. Latter might be easier.

File: sdks/python/apache_beam/io/gcp/bigtableio.py [hunk 2]
```diff
@@ -1,151 +0,0 @@
...
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-#
-
-"""BigTable connector
-
-This module implements writing to BigTable tables.
```
[the remark this answers is not in the document]
Comment (mf2199) [reply sdks/python/apache_beam/io/gcp/bigtableio.py]: If I understand correctly, we are to restore this file in the original version while keeping the altered one in the `experimental` folder. If so, then it's done.

**Comment by aaltay** [comment]
@mf2199 - What is the next step on this PR?

**Comment by param17** [comment]
Can we re-open this? What's the next step?
cc: @mf2199

**Comment by chamikaramj** [comment]
IIRC we could not get BT integration tests working with this PR. Also I think some folks from Google side was going to improve this or look into this but not sure if what work happened. I would like to get this PR to a good working state (including working ITs) before submitting. I think a getting a connector that is not fully fleshed out submitted will do more harm than good.

Another option will be to add a multi-language wrapper for already established Java BT connector which should work for portable runners and Dataflow Runner v2.

**Comment by sachinag** [comment]
@chamikaramj can you see if BT team can get this over the line please?

**Comment by chamikaramj** [comment]
I prefer implementing a x-lang based connector instead of proceeding with this one.

**Pull request closed without merging**